corehq/apps/app_manager/models.py

```diff
@@ -2071,8 +2074,11 @@ class Detail(IndexedSchema, CaseListLookupMixin):
 
     def get_instance_name(self, module):
         value_is_the_default = self.instance_name == 'casedb'
-        if value_is_the_default and module_loads_registry_case(module) or module_uses_inline_search(module):
```

Comment:
Was this a bug before (guessing yes)? Looks like this changes the logic from `if (A and B) or C` to `if A and (B or C)`